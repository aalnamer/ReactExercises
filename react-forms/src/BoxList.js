import React, { useState } from "react";
import { v4 as uuid } from "uuid";
import Box from "./Box";
import NewBoxForm from "./NewBoxForm";

const BoxList = () => {
  const INITIAL_STATE = [];
  const [boxes, setBoxes] = useState(INITIAL_STATE);
  const addBox = (newBox) => {
    setBoxes((boxes) => [...boxes, { ...newBox, id: uuid() }]);
  };
  const remove = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };
  return (
    <div>
      <NewBoxForm addBox={addBox} />
      <div>
        {boxes.map(({ id, width, height, backgroundColor }) => (
          <Box
            id={id}
            width={width}
            height={height}
            backgroundColor={backgroundColor}
            key={id}
            handleRemove={remove}
          />
        ))}
      </div>
    </div>
  );
};

export default BoxList;
