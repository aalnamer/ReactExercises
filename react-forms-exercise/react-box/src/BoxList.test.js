import React from "react";
import { render, fireEvent } from "@testing-library/react";
import BoxList from "./BoxList";

//Smoke
it("renders without crashing", function () {
  render(<BoxList />);
});

//Snapshot
it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

// Use Form

it("should add new box", function () {
  const { queryByText, getByLabelText } = render(<BoxList />);
  const input = getByLabelText("Color");
  const btn = queryByText("Add Box");
  expect(queryByText("Remove This Box!")).not.toBeInTheDocument();
  fireEvent.change(input, { target: { value: "blue" } });
  fireEvent.click(btn);
  expect(queryByText("Remove This Box!")).toBeInTheDocument();
});
