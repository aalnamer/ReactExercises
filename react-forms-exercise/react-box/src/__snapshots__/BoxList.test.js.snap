// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="width"
      >
        New Box
      </label>
      <input
        id="width"
        name="width"
        placeholder="Box Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Box height"
        type="number"
        value=""
      />
      <label
        for="backgroundColor"
      >
        Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Box Color"
        type="text"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
